src/main.rs

```diff
@@ -52,27 +52,27 @@ struct TypeSizeCmd {
 impl TypeSizeCmd {
     fn new(old_args: impl Iterator<Item = String>) -> Self {
         let mut cargo_subcommand = "check";
-        let args = vec![];
-        let mut typesize_args: Vec<String> = vec![];
+        let mut args = vec![];
+        let typesize_args: Vec<String> = vec![];
 
         for arg in old_args {
             match arg.as_str() {
-                "--build" => {
+                "build" => {
                     cargo_subcommand = "build";
                     continue;
                 }
-                "--check" => {
-                    cargo_subcommand = "check";
+                "check" => {
+                    cargo_subcommand = "check --no-run";
```

Comment:
We need the `--no-run` for the `test` subcommand.

Also, we need to push the `"--no-run"` to the `args` list, i.e. `args.push("--no-run".to_string())` instead of including it in the subcommand string.

Otherwise you'll get an error like:
```
error: no such subcommand: `check --no-run`
```

Reply:
Thanks for the catch, updated with the recommended fix.